aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -375,6 +375,10 @@ void TensorIterator::propagate_names_to_outputs() {
 }
 
 void TensorIterator::coalesce_dimensions() {
+  if (!resize_outputs_) {
```

Comment:
This looks like a too strict condition - a lot of operations (copy, advanced indexing, scatter) set `dont_resize_outputs`, and yet they can benefit from coalescing dimensions.  Also, there has been a recent PR fixing what looks to be a similar issue #37788 - can you please check if maybe fixes for these two issues can be combined?

Reply:
Thanks for commenting! I'm checking if we can skip coalescing only for reduction operations. If that still breaks something, I might just add some special handing for `argmin` & `argmax` when the reduced dimension is 1.